R/dependencies.R

```diff
@@ -631,3 +618,62 @@ fileDependencies.Rmd.evaluate <- function(file) {
 
   unique(unlist(deps, recursive = TRUE))
 }
+
+
+
+
+
+fileDependencies.Rmd.tangle <- function(file, encoding = "UTF-8") {
+
+  # discovered packages
+  deps <- list()
+
+  # unique key (line) to split R code with
+  key <- paste0("###--packrat", as.integer(Sys.time()), "\n")
+
+  # rudely override knitr's 'label_code' function so
+  # that we can detect dependencies within inline chunks
+  knitr <- asNamespace("knitr")
+  if (exists("label_code", envir = knitr)) {
+    label_code <- yoink("knitr", "label_code")
+    do.call("unlockBinding", list("label_code", knitr))
+    assign("label_code", function(...) {
+      # paste a known key to the end to split the code chunks with
+      paste0(key, label_code(...))
+    }, envir = knitr)
+
+    on.exit({
+      assign("label_code", label_code, envir = knitr)
+      do.call("lockBinding", list("label_code", knitr))
+    }, add = TRUE)
+  }
+
+  # tangle out file
+  outfile <- tempfile()
+  on.exit({
+    unlink(outfile)
+  }, add = TRUE)
+
+  # attempt to tangle document with our custom hook active
+  x <- knitr::purl(file, output = outfile, quiet = TRUE, documentation = 1L)
```

Comment:
wrap in a try catch like in https://github.com/rstudio/packrat/pull/551/files#diff-a13fefafa2442d59c624b76872450cfcL318